worker/transfer.go

```diff
@@ -50,15 +48,33 @@ func parallelUploadSlab(ctx context.Context, shards [][]byte, hosts []sectorStor
 	defer close(reqChan)
 	respChan := make(chan resp, len(shards))
 	worker := func() {
-		for req := range reqChan {
-			lockID, err := locker.AcquireContract(ctx, req.host.Contract(), contractLockingUploadPriority, 30*time.Second)
-			if err != nil {
-				respChan <- resp{req, types.Hash256{}, err}
-				return
+		for r := range reqChan {
+			doneChan := make(chan struct{})
+			go func(req req) {
```

Comment:
applying the timeout here makes this a little bit messier but it avoids the need for a detached context, it makes it so we do not have to touch the code in `session.go` and it doesn't waste sectors that are uploaded right after the cutoff